### ISSUE TITLE
Feature: KeepAlive and Timeout Support

### ISSUE THREAD

**Issue by vinitparekh17** [DiceDB/dice#632]
Unlike redis this is gonna be important feature for Dice

### Why is it important?
- To Detect dead or unresponsive connections
- To Prevent idle connections from being closed by firewalls or NATs
- To Reduce resource consumption on both client and server sides

### How does it benefits?
- Faster detection of disconnected clients
- Improved reliability in unstable network environments
- Reduced chances of "ghost" connections occupying resources
- 
## 2. Timeout:

- Client timeout
- Command timeout
- Connection timeout

### Why is it important? / How does it benefits?
- Prevent resource exhaustion from slow or hanging connections
- Improve overall system responsiveness

**Comment by Dev79844** [comment]
Hey @vinitparekh17 
I would like to work on this. Can you assign it to me?

**Comment by vinitparekh17** [comment]
@Dev79844 Thanks for showing your interest on this regards, One of these maintainer will assign it to you.

cc: @lucifercr07, @JyotinderSingh

**Comment by lucifercr07** [comment]
@Dev79844 assigned, thanks for contributing.

**Comment by Dev79844** [comment]
Hey
I have planned the implementation of keepalive and timeouts. As soon as the worker implementation is finished, I will add my code for the same.

**Comment by arpitbbhayani** [comment]
Hello @Dev79844,

There has been no activity on this issue for the past 5 days.
It would be awesome if you keep posting updates to this issue so that we know you are actively working on it.

We are really eager to close this issue at the earliest, hence if we continue to see the inactivity, we will have to reassign the issue to someone else. We are doing this to ensure that the project maintains its momentum and others are not blocked on this work.

Just drop a comment with the current status of the work or share any issues you are facing. We can always chip in to help you out.

Thanks again.

**Comment by Dev79844** [comment]
Hey @arpitbbhayani 
Already created a PR few days back. PR #745

**Comment by arpitbbhayani** [comment]
Hello @Dev79844,

There has been no activity on this issue for the past 5 days.
It would be awesome if you keep posting updates to this issue so that we know you are actively working on it.

We are really eager to close this issue at the earliest, hence if we continue to see the inactivity, we will have to reassign the issue to someone else. We are doing this to ensure that the project maintains its momentum and others are not blocked on this work.

Just drop a comment with the current status of the work or share any issues you are facing. We can always chip in to help you out.

Thanks again.

**Comment by Dev79844** [comment]
Hey
Created a new PR #1052 for this issue.

**Comment by arpitbbhayani** [comment]
Hello @Dev79844,

There has been no activity on this issue for the past 5 days.
It would be awesome if you keep posting updates to this issue so that we know you are actively working on it.

We are really eager to close this issue at the earliest, hence if we continue to see the inactivity, we will have to reassign the issue to someone else. We are doing this to ensure that the project maintains its momentum and others are not blocked on this work.

Just drop a comment with the current status of the work or share any issues you are facing. We can always chip in to help you out.

Thanks again.
